commit 35ba785489480a7ac54d3287ef15374c574c03a7
Author: Lorenzo Bianconi <lorenzo.bianconi@redhat.com>
Date:   Mon Jun 4 11:19:41 2018 +0200

    mt76x2: fix mrr idx/count estimation in mt76x2_mac_fill_tx_status()
    
    [ Upstream commit 2d1e9be0016230f3707812243561fbd16f1aea4b ]
    
    Fix mcs and attempt count estimation in mt76x2_mac_fill_tx_status routine
    if the number of tx retries reported by the hw is grater than
    IEEE80211_TX_MAX_RATES
    
    Fixes: 7bc04215a66b ("mt76: add driver code for MT76x2e")
    Signed-off-by: Lorenzo Bianconi <lorenzo.bianconi@redhat.com>
    Signed-off-by: Kalle Valo <kvalo@codeaurora.org>
    Signed-off-by: Sasha Levin <alexander.levin@microsoft.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/net/wireless/mediatek/mt76/mt76x2_mac.c b/drivers/net/wireless/mediatek/mt76/mt76x2_mac.c
index b49aea4da2d6..8985446570bd 100644
--- a/drivers/net/wireless/mediatek/mt76/mt76x2_mac.c
+++ b/drivers/net/wireless/mediatek/mt76/mt76x2_mac.c
@@ -439,15 +439,13 @@ mt76x2_mac_fill_tx_status(struct mt76x2_dev *dev,
 	if (last_rate < IEEE80211_TX_MAX_RATES - 1)
 		rate[last_rate + 1].idx = -1;
 
-	cur_idx = rate[last_rate].idx + st->retry;
+	cur_idx = rate[last_rate].idx + last_rate;
 	for (i = 0; i <= last_rate; i++) {
 		rate[i].flags = rate[last_rate].flags;
 		rate[i].idx = max_t(int, 0, cur_idx - i);
 		rate[i].count = 1;
 	}
-
-	if (last_rate > 0)
-		rate[last_rate - 1].count = st->retry + 1 - last_rate;
+	rate[last_rate].count = st->retry + 1 - last_rate;
 
 	info->status.ampdu_len = n_frames;
 	info->status.ampdu_ack_len = st->success ? n_frames : 0;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1176_35ba785489480a7ac54d3287ef15374c574c03a7.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1176_35ba785489480a7ac54d3287ef15374c574c03a7.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
